File: 3704-count-partitions-with-even-sum-difference/count-partitions-with-even-sum-difference.cs
public class Solution {
    public int CountPartitions(int[] nums) {
        int n = nums.Length, tSum = 0;
    
        foreach (int num in nums) {
            tSum += num;
        }
    
        int left = 0, count = 0;
    
        for (int i = 0; i < n - 1; i++) {
            left += nums[i];
            int right = tSum - left;
        
            if (((left - right) & 1) == 0) {
                count++;
            }
        }
        
        return count;
    }
}

File: 3704-count-partitions-with-even-sum-difference/count-partitions-with-even-sum-difference.py
class Solution:
    def countPartitions(self, nums: List[int]) -> int:
        n = len(nums)
        count = 0

        for i in range(1, n):
            x = nums[0 : i]
            y = nums[i : n]
            xsum = sum(x)
            ysum = sum(y)

            if abs(ysum - xsum) % 2 == 0:
                count += 1

        return count
